workflows/convert-audio/__main__.py

```diff
@@ -0,0 +1,81 @@
+import os
+import sys
+from pathlib import Path
+
+sys.path.insert(1, "workflows/utils")
+
+import luigi
+from list_species import ListSpecies
+from ogg_to_mp3 import OggToMP3
+from pull import Pull
+from push import Push
+
+
+class ConvertAudioTask(luigi.WrapperTask):
+    """Wrapper around the entire DAG."""
+
+    birdclef_root_path = luigi.Parameter()
+
+    text_file_path = luigi.Parameter()
+
+    intermediate_path = luigi.Parameter()
+
+    output_path = luigi.Parameter()
+
+    n_threads = luigi.IntParameter(default=4)
+    parallelism = luigi.IntParameter(default=os.cpu_count())
+
+    def requires(self):
+        list_species_task = ListSpecies(
+            input_path=self.birdclef_root_path,
+            text_output_path=self.text_file_path,
+        )
```

Comment:
It might be instead of having list species as a task, we instead have the GCS client as part of the metatask and only store that state while building up the graph. That means instead of creating a file that has a list of al the species and using that, we have a function that figures out which folders to pull down and generates a DAG for each of those.

Reply:
I could actually be wrong about this: https://luigi.readthedocs.io/en/stable/tasks.html#dynamic-dependencies

Reply:
It looks like idempotency is more important in this case though -- we should be mindful of cases where the encoded file already exists in GCS so we dont go through the process of generating the intermediate files.